packages/location-state-core/docs/API.md

```diff
@@ -1,3 +1,271 @@
 # API
 
+- [State hooks](#State-hooks)
+  - [type `LocationStateDefinition`](#type-LocationStateDefinition)
+  - [`useLocationState`](#uselocationstate)
+  - [`useLocationStateValue`](#uselocationstatevalue)
+  - [`useLocationStateSetter`](#uselocationstatesetter)
+- [Provider](#provider)
+  - [`<LocationStateProvider>`](#Locationstateprovider)
+  - [`createDefaultStores`](#createdefaultstores)
+- [Syncer](#syncer)
+  - [`NavigationSyncer`](#navigationsyncer)
+- [Store](#store)
+  - [`StorageStore`](#storagestore)
+  - [`URLStore`](#urlstore)
+  - [custom `Store`](#custom-store)
+  - [`getHooksWith`](#gethookswith)
+
+## State hooks
+
+State hooks は [`<LocationStateProvider>`](#Locationstateprovider) の子孫コンポーネントでのみ利用可能です。
+
+### type `LocationStateDefinition`
+
+hooks の共通オプションは`LocationStateDefinition`として、以下のように定義されています。
+
+```ts
+type LocationStateDefinition<
+  T,
+  StoreName extends string = "session" | "url",
+> = {
+  name: string;
+  defaultValue: T;
+  storeName: StoreName;
+  refine?: Refine<T>;
+};
+```
+
+- `name`: state を一意に判別する名前
+- `defaultValue`: state のデフォルト値
+- `storeName`: state の保存先。`session`と`url`の 2 つが利用可能（カスタマイズ可能）
+- `refine?`: state 復元時に検証・変換する関数。[Refine](#type-refine)を参照
+
+### type `Refine`
+
+値を復元する時に検証・変換を行う関数です。`undefined`を返すとデフォルト値となります。
+
+```ts
+type Refine<T> = (value: unknown) => T | undefined;
+```
+
+- `value`: 復元を試みる値
+
+#### Example
+
+```ts
+const zodRefine =
+  <T extends unknown>(schema: ZodType<T>): Refine<T> =>
+  (value) => {
+    const result = schema.safeParse(value);
+    return result.success ? result.data : undefined;
+  };
+
+const [counter, setCounter] = useLocationState({
+  name: "counter",
+  defaultValue: 0,
+  storeName,
+  refine: zodRefine(z.number()),
+});
+```
+
+### `useLocationState`
+
+オプションを指定して`useLocationState`を呼び出すことで、`useState`同様 state と setter を返します。
+
+```ts
+declare const useLocationState: <T>(
+  definition: LocationStateDefinition<T, DefaultStoreName>,
+) => [T, SetState<T>];
+```
+
+#### Parameters
+
+- `definition`: [`LocationStateDefinition`](#type-locationstatedefinition)を参照
+
+#### Returns
+
+state と state の更新関数を返します。
```

Comment:
ここはちゃんと配列を返すことを説明したい (型定義で分かるとはいえ)
Reactなら (少しくどい気もするけど)

> useState returns an array with exactly two values:
> 1.  The current state. During the first render, it will match the initialState you have passed.
> 2.  The [set function](https://react.dev/reference/react/useState#setstate) that lets you update the state to a different value and trigger a re-render.

Recoilは1行説明のところで

> Returns a tuple where the first element is the value of state and the second element is a setter function that will update the value of the given state when called.

Reply:
ちょっと短いですが、Recoilっぽく書いてみました。
76b55715574a2cadeb03f63961b162c6bc1fe113